hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/AccountReadableKVState.java

```diff
@@ -0,0 +1,243 @@
+    private final TokenAllowanceRepository tokenAllowanceRepository;
+    private final CryptoAllowanceRepository cryptoAllowanceRepository;
+    private final TokenAccountRepository tokenAccountRepository;
+    private final AccountBalanceRepository accountBalanceRepository;
+
+    public AccountReadableKVState(
+            CommonEntityAccessor commonEntityAccessor,
+            NftAllowanceRepository nftAllowanceRepository,
+            NftRepository nftRepository,
+            TokenAllowanceRepository tokenAllowanceRepository,
+            CryptoAllowanceRepository cryptoAllowanceRepository,
+            TokenAccountRepository tokenAccountRepository,
+            AccountBalanceRepository accountBalanceRepository) {
+        super(KEY);
+        this.commonEntityAccessor = commonEntityAccessor;
+        this.nftAllowanceRepository = nftAllowanceRepository;
+        this.nftRepository = nftRepository;
+        this.tokenAllowanceRepository = tokenAllowanceRepository;
+        this.cryptoAllowanceRepository = cryptoAllowanceRepository;
+        this.tokenAccountRepository = tokenAccountRepository;
+        this.accountBalanceRepository = accountBalanceRepository;
+    }
+
+    @Override
+    protected Account readFromDataSource(@Nonnull AccountID key) {
+        final var timestamp = ContractCallContext.get().getTimestamp();
+        return commonEntityAccessor
+                .get(key, timestamp)
+                .map(entity -> accountFromEntity(entity, timestamp))
+                .orElse(null);
+    }
+
+    @Override
+    protected @Nonnull Iterator<AccountID> iterateFromDataSource() {
+        return Collections.emptyIterator();
+    }
+
+    @Override
+    public long size() {
+        return 0;
+    }
+
+    private Account accountFromEntity(Entity entity, final Optional<Long> timestamp) {
+        var tokenAccountBalances = getNumberOfAllAndPositiveBalanceTokenAssociations(entity.getId(), timestamp);
+
+        return Account.newBuilder()
+                .accountId(convertCanonicalAccountIdFromEntity(entity))
+                .alias(
+                        entity.getEvmAddress() != null && entity.getEvmAddress().length > 0
+                                ? Bytes.wrap(entity.getEvmAddress())
+                                : Bytes.EMPTY)
+                .key(parseKey(entity.getKey()))
+                .expirationSecond(TimeUnit.SECONDS.convert(entity.getEffectiveExpiration(), TimeUnit.NANOSECONDS))
+                .tinybarBalance(getAccountBalance(entity, timestamp))
+                .memo(entity.getMemo())
+                .deleted(Optional.ofNullable(entity.getDeleted()).orElse(false))
+                .receiverSigRequired(entity.getReceiverSigRequired() != null && entity.getReceiverSigRequired())
+                .numberOwnedNfts(getOwnedNfts(entity.getId(), timestamp))
+                .maxAutoAssociations(Optional.ofNullable(entity.getMaxAutomaticTokenAssociations())
+                        .orElse(0))
```

Comment:
Avoid allocating temp Optional. Also change deleted()

```suggestion
                .maxAutoAssociations(Objects.requireNonNullElse(entity.getMaxAutomaticTokenAssociations(), 0))
```